cyclepainter.py

```diff
@@ -183,15 +217,25 @@ def integrate(self, omega):
 
     def intersection_number(self, path, eps=1e-10):
         ''' Calculates the intersection number with another path. '''
+        # We first just state identify that the self-inersection of any number is 0
         if path == self:
             return 0
+        # Alternatively we want to sum over all the intersections signed by orientation.
+        # We will run into problems if the intersection is at a branch cut for two reasons:
+        # 1) This is not a real intersection, it is an artefact of the method
+        # 2) The method for finding intersection points and orientations is not wll defined for colinear points. 
+        # Given the sensitivity of the input method in cyclepainter, we should only worry about this when 
+        # taking the intersection of a cycle with the same path but shifted by origin sheet. 
+        # Here, previously cyclepainter was detecting the artefact intersection depending on orientation.
+        # Now we robustly detect these, we need to know they are artefacts caused by branch cut. 
+        # Here we use the genericity of the prblem to assume all these endpoint intersections are artefacts. 
```

Comment:
Great! This indeed looks like a good solution which (if it works as expected) makes Cyclepainter much more reliable

Reply:
Yep there are obviously cases one can imagine where this will fail, but I believe that because of the resolution in drawing paths we can only expect them when we are certain paths are colinear, or when we are certain they intersect on a cut, i.e. they are produce by cyclepainter cutting paths that cross branch cuts.